OpenRA.Mods.Common/Traits/Conditions/GrantChargedConditionOnToggle.cs

```diff
@@ -35,38 +35,47 @@ public class GrantConditionOnDeployWithChargeInfo : PausableConditionalTraitInfo
 		[Desc("Cooldown (in ticks) to reach full charge.")]
 		public readonly int ChargeDuration = 500;
 
-		[Desc("The amount of charge that needs to be present for deploy to be issued. " +
+		[Desc("The amount of charge that needs to be present to turn on the condition. " +
 			"If set to -1, threshold is set to full charge. " +
 			"If activated without full charge " + nameof(ConditionDuration) + " is percentally smaller.")]
 		public readonly int ChargeThreshhold = -1;
 
 		[Desc("How long (in ticks) should the condition stay active?")]
 		public readonly int ConditionDuration = 1;
 
-		[Desc("Can " + nameof(DeployedCondition) + " be canceled by followup deploy order?")]
+		[Desc("Can " + nameof(ActivatedCondition) + " be canceled turned off manually?")]
 		public readonly bool CanCancelCondition = false;
 
+		[Desc("Should we interrupt the current activity and wait for it to finish?")]
+		public readonly bool CancelsCurrentOrders = false;
```

Comment:
What is the use case of this flag? We have `GrantConditionOnDeploy` for things that "deploy" (i.e. cancel current orders). It would be better to keep the "OnDeploy" terminology for that use case and keep this trait separate.

Reply:
grant condition on deploy doesn't do charge, the usecase is the general flexibility conditions allow to modders. The condition may trigger something that needs the unit to be stationary

Reply:
Charge-vs-not and Deploy-vs-activate are orthogonal features. The timed-condition-on-deploy feature should be either a feature of `GrantConditionOnDeploy` or a separate `GrantChargedConditionOnDeploy` trait.